py/qstr.c

```diff
@@ -141,14 +141,19 @@ STATIC qstr qstr_add(const byte *q_ptr) {
 
     // make sure we have room in the pool for a new qstr
     if (MP_STATE_VM(last_pool)->len >= MP_STATE_VM(last_pool)->alloc) {
-        qstr_pool_t *pool = m_new_obj_var_maybe(qstr_pool_t, const char*, MP_STATE_VM(last_pool)->alloc * 2);
+        size_t new_alloc = MP_STATE_VM(last_pool)->alloc * 2;
+        #ifdef MICROPY_QSTR_EXTRA_POOL
+        // Put a lower bound on the allocation size in case the extra qstr pool has few entries
+        new_alloc = MAX(10, new_alloc);
```

Comment:
10 seems to be pretty arbitrary. How does it interact with default alloc block size?

Reply:
10 was chosen to match how it behaves without an extra qstr pool, see line 101 above.  Maybe that can become a simple macro, to make the similarity it explicit.

Probably it would be good to provide global config macros for this initial value, and also for the way allocation increases (it doubles at the moment), but that's out of scope of this PR.